tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,330 @@
+package clusterroletemplatebindings
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type CRTBGenTestSuite struct {
+	suite.Suite
+	client            *rancher.Client
+	crtbName          string
+	roleTemplateName  string
+	userPrincipalName string
+	session           *session.Session
+	cluster           *management.Cluster
+	clusterName       string
+	remoteClusterName string
+	clusterID         string
+	remoteClusterID   string
+	clusterV1         *v1.Cluster
+}
+
+const (
+	crtbJSONFilePath = "resources/createCRTBTemplate.json"
+)
+
+func (crtb *CRTBGenTestSuite) TearDownSuite() {
+	crtb.session.Cleanup()
+}
+
+func (crtb *CRTBGenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	crtb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(crtb.T(), err)
+
+	crtb.client = client
+	crtb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(crtb.T(), crtb.clusterName, "Cluster name to install should be set")
+
+	crtb.clusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.clusterName)
+	require.NoError(crtb.T(), err, "Error getting cluster ID")
+	crtb.cluster, err = crtb.client.Management.Cluster.ByID(crtb.clusterID)
+	require.NoError(crtb.T(), err)
+
+	crtb.remoteClusterName = getNonLocalCluster(crtb.T(), client)
+	crtb.remoteClusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.remoteClusterName)
+	require.NoError(crtb.T(), err)
+}
+
+func getNonLocalCluster(t *testing.T, client *rancher.Client) string {
+	clusterList, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(t, err)
+	var clusterName string
+	for _, clusterData := range clusterList.Data {
+		if clusterData.JSONResp["metadata"] != nil {
+			metadata, ok := clusterData.JSONResp["metadata"].(map[string]interface{})
+			if ok {
+				if nameValue, exists := metadata["name"]; exists {
+					if nameStr, isString := nameValue.(string); isString && nameStr != "local" {
+						clusterName = nameStr
+						break
+					}
+				}
+			}
+		}
+	}
+	return clusterName
+}
+
+func validateCrtbCreationBySteveWithAllRequiredInputValues(t *testing.T, client *rancher.Client, remoteClusterID string, roleTemplateName string) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
```

Comment:
Can we add a require.NoError after creating a user to check if the user that this test depends on is indeed created? This applies to all tests.

Reply:
Please check now, its resolved